framework/behaviors/AttributeBehavior.php

```diff
@@ -118,6 +123,10 @@ public function evaluateAttributes($event)
             foreach ($attributes as $attribute) {
                 // ignore attribute names which are not string (e.g. when set by TimestampBehavior::updatedAtAttribute)
                 if (is_string($attribute)) {
+                    // skip attribute update when it is not empty and the corresponding mode is enabled
+                    if ($this->skipUpdateOnFilled && !empty($this->owner->$attribute)) {
```

Comment:
@dynasource I used the `skipUpdateOn...` property because it is consistent with the `skipUpdateOnClean`. Do you really insist on naming the property inconsistently even if it serves the similar purpose? Maybe we can use `skipUpdateOnNonEmpty`?

If you insist on naming it differently I can suggest something like `preserveExisting`.

What property name would you suggest?

>  When is it about a change or about being empty?
It's about being empty.